File: chapter11/listing_11_21-todo-scala-customized-sourceset/model/src/scala/com/manning/gia/todo/model/ToDoItem.scala
package com.manning.gia.todo.model

class ToDoItem extends Ordered[ToDoItem] {
    var id: Long = _
    var name: String = _
    var completed: Boolean = _
    
    def getId: Long = {
        id
    }
    
    def getName: String = {
        name
    }
    
    def isCompleted: Boolean = {
        completed
    }
    
    def setId(id: Long) = {
        this.id = id
    }
    
    def setName(name: String) = {
        this.name = name
    }
    
    def setCompleted(completed: Boolean) = {
        this.completed = completed
    }
    
    override def compare(that: ToDoItem) = this.id compare that.id

    override def equals(that: Any) = {
        that match {
            case t: ToDoItem => t.id == id && t.name == name && t.completed == completed
            case _ => false
        }
    }
}

File: chapter03/todo-app-external-dependency/src/main/java/com/manning/gia/todo/model/ToDoItem.java
package com.manning.gia.todo.model;

public class ToDoItem implements Comparable<ToDoItem> {
    private Long id;
    private String name;
    private boolean completed;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isCompleted() {
        return completed;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ToDoItem that = (ToDoItem) o;

        if (completed != that.completed) return false;
        if (id != null ? !id.equals(that.id) : that.id != null) return false;
        if (name != null ? !name.equals(that.name) : that.name != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id != null ? id.hashCode() : 0;
        result = 31 * result + (name != null ? name.hashCode() : 0);
        result = 31 * result + (completed ? 1 : 0);
        return result;
    }

    @Override
    public int compareTo(ToDoItem toDoItem) {
        return this.getId().compareTo(toDoItem.getId());
    }

    @Override
    public String toString() {
        return id + ": " + name + " [completed: " + completed + "]";
    }
}